test/e2e/network/topology_hints.go

```diff
@@ -83,6 +84,33 @@ var _ = common.SIGDescribe("Feature:Topology Hints", func() {
 		framework.ExpectNoError(err, "timed out waiting for DaemonSets to be ready")
 
 		nodeNames := e2edaemonset.SchedulableNodes(c, ds)
+
+		// All Nodes should have same allocatable CPUs. If not, then skip the test.
+		nodeNamesMap := map[string]bool{}
+		for _, nodeName := range nodeNames {
+			nodeNamesMap[nodeName] = true
+		}
+		nodeList, err := c.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
+		framework.ExpectNoError(err, "Error when listing all Nodes")
+		var lastNodeCPU resource.Quantity
+		firstNode := true
+		for _, node := range nodeList.Items {
+			if !nodeNamesMap[node.Name] {
+				continue
+			}
+			nodeCPU, found := node.Status.Allocatable[v1.ResourceCPU]
+			if !found {
+				framework.Failf("Error when getting allocatable CPU of Node '%s'", node.Name)
+			}
+			if firstNode {
+				lastNodeCPU = nodeCPU
+				firstNode = false
+			} else if !nodeCPU.Equal(lastNodeCPU) {
+				e2eskipper.Skipf("Not all Nodes have same allocatable CPUs. Value of Node '%s' is different from the last one. %d not equals %d",
```

Comment:
Maybe reword this to clarify what we expected here.
```suggestion
				e2eskipper.Skipf("Expected Nodes to have equivalent allocatable CPUs, but Node '%s' is different from the previous one. %d not equals %d",
```

Reply:
done